persistence-api/src/main/java/io/stargate/db/query/builder/QueryBuilderImpl.java

```diff
@@ -0,0 +1,1675 @@
+import io.stargate.db.schema.Schema;
+import io.stargate.db.schema.SchemaEntity;
+import io.stargate.db.schema.SecondaryIndex;
+import io.stargate.db.schema.Table;
+import io.stargate.db.schema.UserDefinedType;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import javax.annotation.Nullable;
+import org.javatuples.Pair;
+
+/** Convenience builder for creating queries. */
+@GenerateEmbeddedDSL(
+    modifier = AFModifier.DEFAULT,
+    autoVarArgs = false,
+    name = "QueryBuilder",
+    syntax = "(<keyspace>|<table>|<insert>|<update>|<delete>|<select>|<index>|<type>) build",
+    where = {
+      @SubExpr(
+          name = "keyspace",
+          definedAs = "<keyspaceCreate> | <keyspaceAlter> | (drop keyspace ifExists?)"),
+      @SubExpr(
+          name = "keyspaceCreate",
+          definedAs = "(create keyspace ifNotExists? withReplication andDurableWrites?)"),
+      @SubExpr(
+          name = "keyspaceAlter",
+          definedAs = "(alter keyspace (withReplication andDurableWrites?)?)"),
+      @SubExpr(
+          name = "table",
+          definedAs =
+              "(create table ifNotExists? column+ withComment? (<withVL>|<withEL>)? withDefaultTTL?) | (alter table (((addColumn+)? | (dropColumn+)? | (renameColumn+)?) | (withComment? withDefaultTTL? (<withVL>|<withEL>|<withoutVL>|<withoutEL>)?))) | (drop table ifExists?) | (truncate table) | (<renameGraphLabel>)"),
+      @SubExpr(name = "type", definedAs = "(create type ifNotExists?)|(drop type ifExists?)"),
+      @SubExpr(name = "insert", definedAs = "insertInto value+ ifNotExists? ttl? timestamp?"),
+      @SubExpr(name = "update", definedAs = "update ttl? timestamp? value+ where+ ifs* ifExists?"),
+      @SubExpr(name = "delete", definedAs = "delete column* from timestamp? where+ ifs* ifExists?"),
+      @SubExpr(
+          name = "select",
+          definedAs =
+              "select star? column* writeTimeColumn? from (where* limit? orderBy*) allowFiltering?"),
+      @SubExpr(
+          name = "index",
+          definedAs =
+              "(drop ((materializedView|index) ifExists?)) | (create ((materializedView ifNotExists? asSelect (column+) from)"
+                  + " | (custom? index ifNotExists? on column (indexKeys|indexValues|indexEntries|indexFull|indexingType)?)))"),
+      @SubExpr(name = "withVL", definedAs = "withVertexLabel"),
+      @SubExpr(
+          name = "withEL",
+          definedAs = "withEdgeLabel fromVertexLabel fromColumn+ toVertexLabel toColumn+"),
+      @SubExpr(name = "withoutVL", definedAs = "withoutVertexLabel"),
+      @SubExpr(name = "withoutEL", definedAs = "withoutEdgeLabel"),
+      @SubExpr(
+          name = "renameGraphLabel",
+          definedAs = "alter table rename (vertexLabel|edgeLabel) fromLabel? toLabel"),
```

Comment:
I know this code existed before, but since we're remaking the query API in this PR, do we intend to support Graph-specific queries? AFAIK, they will only work in DSE 6.8?

Reply:
No real clue. I hesitated removing it, but didn't wanted to make it sound like I was taking a stand on graph :).

But it's not used, and it's easy enough to add back later, so since you pointed it out, I'll remove.